.eleventy.js

```diff
@@ -115,35 +115,32 @@ module.exports = function (eleventyConfig) {
 	eleventyConfig.addPassthroughCopy("src/img");
 	eleventyConfig.addPassthroughCopy("./CNAME");
 	// eleventyConfig.addPassthroughCopy("src/.gitignore");
-	eleventyConfig.addPassthroughCopy({ "dinky/assets/js": "assets/js" });
-	eleventyConfig.addPassthroughCopy({
-		"dinky/assets/images": "assets/images",
-	});
-	eleventyConfig.addPassthroughCopy({
-		"dinky/_sass": "sass/dinky/_sass",
-	});
+	// eleventyConfig.addPassthroughCopy({ "dinky/assets/js": "assets/js" });
+	// eleventyConfig.addPassthroughCopy({
+	// 	"dinky/assets/images": "assets/images",
+	// });
+	// eleventyConfig.addPassthroughCopy({
+	// 	"dinky/_sass": "sass/dinky/_sass",
+	// });
 	eleventyConfig.addPassthroughCopy({
 		"src/_sass": "sass/src/_sass",
 	});
 
-	const pathNormalizer = function(pathString){
-		return normalize(path.normalize(path.resolve(".")))
+const njkEngine = require("nunjucks").configure(
+	[
+		path.join(siteConfiguration.dir.input, siteConfiguration.dir.includes),
+		path.join(siteConfiguration.dir.input, siteConfiguration.dir.layouts),
+		siteConfiguration.dir.input,
+	],
+	{
+		autoescape: false,
+		throwOnUndefined: true
 	}
+);
 
-	// Nunjucks Filters
-	let nunjucksEnvironment = new Nunjucks.Environment(
-		new Nunjucks.FileSystemLoader([
-			pathNormalizer(siteConfiguration.dir.includes),
-			pathNormalizer(siteConfiguration.dir.input),
-			pathNormalizer(".")
-		]),
-		{
-			throwOnUndefined: throwOnUndefinedSetting,
-			autoescape: true
-		}
-	);
-	eleventyConfig.setLibrary("njk", nunjucksEnvironment);
-	eleventyConfig.addNunjucksFilter("interpolate", function(value) {
+eleventyConfig.setLibrary('njk', njkEngine );  //: autoescape for CSS rules
```

Comment:
Inspired by another 11ty issue that you commented on, but I'm sure the `Nunjucks.FileSystemLoader` approach works just as well.

Reply:
@pdehaan I was able to get everything to work with your changes and this configuration but.... the thing I don't understand is why the other way with `new Nunjucks.FileSystemLoader(` doesn't seem to work. 

```javascript
let nunjucksEnvironment = Nunjucks.configure(
		new Nunjucks.FileSystemLoader([
                    '/Users/username/Dev/devblog/src/_includes',
                    '/Users/username/Dev/devblog/src/_layouts',
                    '/Users/username/Dev/devblog/src',
                    '.'
                  ]),
		  {
			  throwOnUndefined: throwOnUndefinedSetting,
			  autoescape: true,
		  }
	  );
eleventyConfig.setLibrary("njk", nunjucksEnvironment);
```

This matches what I've seen elsewhere and the given documentation, but I tried it with absolute and relative paths, and it didn't load the folders properly. 

This should work right? But it doesn't seem to read the paths. The site builds fine this way, but I am somewhat disappointed I couldn't get it working the original way, or figure out why it isn't working.

Reply:
I might not be following. I recloned your repo and built it locally and it seems to work for me (well, except for the Sass stuff, I had to comment out Sass in .eleventy.js):

```diff
diff --git a/.eleventy.js b/.eleventy.js
index 6dfc67b..07f4d19 100644
--- a/.eleventy.js
+++ b/.eleventy.js
@@ -99,11 +99,11 @@ module.exports = function (eleventyConfig) {
 	});
 	eleventyConfig.addWatchTarget("./_custom-plugins/");
 	eleventyConfig.addWatchTarget("./src/_sass");
-	sassBuild(domain_name);
+	// sassBuild(domain_name);
 	eleventyConfig.on("beforeWatch", (changedFiles) => {
 		// changedFiles is an array of files that changed
 		// to trigger the watch/serve build
-		sassBuild(domain_name);
+		// sassBuild(domain_name);
 	});
 
 	// https://www.npmjs.com/package/@quasibit/eleventy-plugin-sitemap
@@ -178,19 +178,24 @@ module.exports = function (eleventyConfig) {
 	);
 	eleventyConfig.setLibrary("njk", nunjucksEnvironment);
     */
-	const njkEngine = require("nunjucks").configure(
-		[
-			path.join(
-				siteConfiguration.dir.input,
-				siteConfiguration.dir.includes
-			),
-			path.join(
-				siteConfiguration.dir.input,
-				siteConfiguration.dir.layouts
-			),
+
+	const njkConfig = [
+		path.join(
+			siteConfiguration.dir.input,
+			siteConfiguration.dir.includes
+		),
+		path.join(
 			siteConfiguration.dir.input,
-			normalize(path.normalize(".")),
-		],
+			siteConfiguration.dir.layouts
+		),
+		siteConfiguration.dir.input,
+		normalize(path.normalize(".")),
+	];
+
+	console.log("other nunjucksFileSystem", njkConfig);
+
+	const njkEngine = require("nunjucks").configure(
+		njkConfig,
 		{
 			autoescape: false,
 			throwOnUndefined: true,
@@ -198,12 +203,7 @@ module.exports = function (eleventyConfig) {
 	);
 	// /Users/zuckerscharffa/Dev/fightwithtooldev/.eleventy.js
 	// /Users/zuckerscharffa/Dev/fightwithtooldev/src
-	console.log("other nunjucksFileSystem", [
-		path.join(siteConfiguration.dir.input, siteConfiguration.dir.includes),
-		path.join(siteConfiguration.dir.input, siteConfiguration.dir.layouts),
-		siteConfiguration.dir.input,
-		normalize(path.normalize(".")),
-	]);
+
 	eleventyConfig.setLibrary("njk", njkEngine); //: autoescape for CSS rules
 	//eleventyConfig.addNunjucksFilter("interpolate", function(value) {
 	//	return Nunjucks.renderString(text, this.ctx);
```

### OUTPUT

```sh
npm run build -- --quiet

> fightwithtooldev@1.0.0 build
> eleventy "--quiet"

other nunjucksFileSystem [ 'src/_includes', 'src/_layouts', 'src', '.' ]
(node:81014) [DEP0128] DeprecationWarning: Invalid 'main' field in '/private/tmp/devblog/node_modules/emitter-mixin/package.json' of 'y'. Please either fix that or report it to the module author
(Use `node --trace-deprecation ...` to show where the warning was created)
[ 'backreads', 'devblog' ]
pagedPosts Starters
Copied 6 files / Wrote 47 files in 0.67 seconds (14.3ms each, v0.12.1)
```

And I think the `{% include %}` was working as expected. It built fine, and the tag pages seem to render like I thought (since that might have been the only `include` statement I found:

```html
<h2>Posts tagged: GPC</h2>
<div id="post-summary-list">
    <ul>
		<li><div>
    <h3> <a href="/posts/projects/devblog/hello-day-20/">Hello World Devblog - Pt. 20</a></h3>
    	<p>More devblog</p>
		<p><a href="/posts/projects/devblog/hello-day-20/">Read more...</a></p>
</div>
</li>
		<li><div>
    <h3> <a href="/posts/projects/devblog/hello-day-19/">Hello World Devblog - Pt. 19</a></h3>
    	<p>More devblog</p>
		<p><a href="/posts/projects/devblog/hello-day-19/">Read more...</a></p>
</div>
</li>
    </ul>
</div>
```

via 

```njk
{% block content %}
<h2>Posts tagged: {{paged.tagName}}</h2>
<div id="post-summary-list">
    <ul>
	{%- for post in paged.posts %}
		<li>{% include "partials/post-summary.njk" %}</li>
	{%- endfor %}
    </ul>
</div>
{% endblock %}
```